src/Sentry.AzureFunctions.Worker/SentryFunctionsWorkerMiddleware.cs

```diff
@@ -0,0 +1,81 @@
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Middleware;
+
+namespace Sentry.AzureFunctions.Worker;
+
+internal class SentryFunctionsWorkerMiddleware : IFunctionsWorkerMiddleware
+{
+    private readonly IHub _hub;
+
+    public SentryFunctionsWorkerMiddleware(IHub hub)
+    {
+        _hub = hub;
+    }
+
+    public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next)
+    {
+        try
+        {
+            var transactionName = await GetTransactionNameAsync(context) ?? context.FunctionDefinition.Name;
+
+            _hub.ConfigureScope(scope =>
+            {
+                var transaction = _hub.StartTransaction(transactionName, "function");
+                scope.Transaction = transaction;
+
+                context.CancellationToken.Register(() => scope.SetExtra("aborted", true));
```

Comment:
I think this is unnecessary, since we're already setting the transaction aborted here:

https://github.com/getsentry/sentry-dotnet/blob/b1bfe1efc04eb4c911a85f1cf4cd2e5a176d7c8a/src/Sentry/Internal/Hub.cs#L360-L366

That's presuming a `OperationCanceledException` (or `TaskCanceledException`) is already being thrown during cancellation.  If not, then I think we should just do `context.CancellationToken.ThrowIfCancellationRequested();` here, rather than registering.

Reply:
I'm assuming we still want to start a transaction and then call `context.CancellationToken.ThrowIfCancellationRequested();` so Sentry captures it as part of the transaction. Right, @mattjohnsonpint?

Reply:
Yes, exactly.  I just meant registering for cancellation was probably unnecessary.